### PR TITLE
Response Formatting

### PR DESCRIPTION
**Pull request by champion-chenpeng**
Screenshot 2024-02-29 at 3 21 11 PM

### CODE REVIEW

File: frontend/src/components/ChatMessage.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import bot from '../media/img/icon-bot.png'
 import user from '../media/img/icon-user.png'
+import ChatbotLoading from './ChatbotLoading';
```
Comment (katiehsieh) [inline frontend/src/components/ChatMessage.js:3]: Is there another ChatbotLoading component file that needs to be committed?

Comment (champion-chenpeng) [reply frontend/src/components/ChatMessage.js:3]: Sorry, I forgot to add it.

**Comment by champion-chenpeng** [comment]
image

It seems that the sources returned from backend are repeated.

**Comment by champion-chenpeng** [comment]
image

The formatting works well.

**Review by katiehsieh — APPROVED** [review]
Looks great!

File: frontend/src/components/ChatMessage.js [hunk 2]
```diff
@@ -1,14 +1,49 @@
...
+  const htmlString = parse(props.text);
+  return (
+    <div>
+      <div dangerouslySetInnerHTML={{ __html: htmlString }} />
+      <hr />
+      {props.sources.map(source => {
+        const prefix = "https://github.com/CSE210-TEAM2/team2-chatbot/tree/main/backend/";
+        const rp = source.metadata.source;
+        const pageNumber = source.metadata.page;
+        return <li><a href={prefix + rp}>{rp + ", P" + pageNumber}</a></li>;
```
[the remark this answers is not in the document]
Comment (champion-chenpeng) [reply frontend/src/components/ChatMessage.js:18]: Good suggestion.

**Pull request merged**